src/librustc_mir/transform/qualify_consts.rs

```diff
@@ -1067,6 +1100,10 @@ impl<'a, 'tcx> Visitor<'tcx> for Qualifier<'a, 'tcx, 'tcx> {
 
             // Deny *any* live drops anywhere other than functions.
             if self.mode != Mode::Fn {
+                if self.tcx.features().unleash_the_miri_inside_of_you {
+                    self.tcx.sess.span_warn(self.span, "skipping const checks");
+                    return;
+                }
```

Comment:
That's an awful lot of repetition (9th time this exact same thing occurs...) which seems ungreat code quality and maintenance wise... we can't use a lambda due to `return` but can we use a macro or something?